maintainers/scripts/add-test-version.sh

```diff
@@ -0,0 +1,62 @@
+#! /usr/bin/env bash
+
+set -euo pipefail
+
+info() {
+    echo "INFO:" "$@" > /dev/stderr
+}
+
+warn() {
+    echo "WARNING:" "$@" > /dev/stderr
+}
+
+error () {
+    echo "ERROR:" "$@" > /dev/stderr
+    exit 1
+}
+
+if [ $# == 0 ]; then
+    # git checkout pkgs
+    ALL_PKGS="nix-instantiate --eval -E 'with import ./. {}; lib.attrNames pkgs' --json | jq '.[]' -r"
+    exec nix-shell -p bash ripgrep ag nixpkgs-fmt --run "$ALL_PKGS | xargs -n1 -P4 -- bash $0"
+fi
+
+if [ $# -ne 1 ]; then
+    error "Wrong number of arguments"
+fi
+
+PNAME=$1
+FILE=$(EDITOR=echo nix edit "(import ./. {}).$PNAME")
+info $FILE
+
+# No multi derivation files
+[ $(rg "mkDerivation" $FILE | wc -l) == 1 ]
+
+# No files that already use testVersion
+! rg "[^\w]testVersion[^\w]" $FILE
+rg "meta" $FILE
+
+# We want files that are already properly formatted, so that we can reformat later.
+nixpkgs-fmt --check $FILE
+
+cleanup () {
+    while true; do
+        git checkout $FILE 2> /dev/null && break
+        sleep 0.5
+    done
+    info "cleanup complete"
+}
+
+trap cleanup EXIT
+
+# We start making changes below this line. Undo them when needed.
+
+# Add the testVersion argument to the main file
+sed -zi "0,/{/{s/{\(.*\)}:/{\1, testVersion, $PNAME}:/}" $FILE
+sed -i "s/meta =/passthru.tests.version = testVersion { package = $PNAME; };\n\n meta =/" $FILE
+
+nixpkgs-fmt $FILE
+timeout 120 nix-build . -A $PNAME.tests.version
```

Comment:
```suggestion
timeout 60 nix-build . -A $PNAME.tests.version
```

Reply:
I have it at 120 to allow some time for nix-build to start up and actually build/fetch the package.